packages/react/src/components/Tabs/next/Tabs.js

```diff
@@ -637,18 +637,56 @@ const TabPanel = React.forwardRef(function TabPanel(
   const ref = useMergedRefs([forwardRef, panel]);
 
   const [tabIndex, setTabIndex] = useState('0');
+  const [interactiveContent, setInteractiveContent] = useState(false);
   const { selectedIndex, baseId } = React.useContext(TabsContext);
   const index = React.useContext(TabPanelContext);
   const id = `${baseId}-tabpanel-${index}`;
   const tabId = `${baseId}-tab-${index}`;
-  const className = cx(`${prefix}--tab-content`, customClassName);
+  const className = cx(`${prefix}--tab-content`, customClassName, {
+    [`${prefix}--tab-content--interactive`]: interactiveContent,
+  });
+
+  useEffectOnce(() => {
+    if (!panel.current) {
+      return;
+    }
+
+    const content = getInteractiveContent(panel.current);
+    if (content) {
+      setInteractiveContent(true);
+      setTabIndex('-1');
+    }
+  });
 
   // tabindex should only be 0 if no interactive content in children
   useEffect(() => {
-    const interactiveContent = getInteractiveContent(panel.current);
-    if (interactiveContent) {
-      setTabIndex('-1');
+    if (!panel.current) {
+      return;
     }
+
+    const { current: node } = panel;
+
+    function callback() {
+      const content = getInteractiveContent(node);
+      if (content) {
+        setInteractiveContent(true);
+        setTabIndex('-1');
+      }
+
+      setInteractiveContent(false);
+      setTabIndex('0');
+    }
```

Comment:
Should this be in an `else` block? Looking at this in the diff view, this seems like it would set interactive content / tab index if content is truthy and then would set it to false/0 afterwards